server/src/test/java/org/elasticsearch/gateway/MetaStateServiceTests.java

```diff
@@ -21,83 +21,163 @@
 import org.elasticsearch.Version;
 import org.elasticsearch.cluster.metadata.IndexMetaData;
 import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.cluster.metadata.MetaState;
+import org.elasticsearch.common.UUIDs;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.env.NodeEnvironment;
-import org.elasticsearch.index.Index;
 import org.elasticsearch.test.ESTestCase;
 
 import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.nullValue;
 
 public class MetaStateServiceTests extends ESTestCase {
```

Comment:
I think these tests should assert that the `MetaStateService` is in its reset state after writing the manifest file. Maybe they should also sometimes construct a new `MetaStateService` when reading things back in.

Reply:
++

Reply:
https://github.com/elastic/elasticsearch/pull/35199/commits/15b505969b3253d59f31f51854a9ccdc1aed8df3